x/compliance/genesis_test.go

```diff
@@ -0,0 +1,381 @@
+package compliance_test
+
+import (
+	"sort"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/status-im/keycard-go/hexutils"
+	"github.com/stretchr/testify/require"
+
+	testkeeper "swisstronik/testutil/keeper"
+	"swisstronik/x/compliance"
+	"swisstronik/x/compliance/types"
+)
+
+func TestInitGenesis_Validation(t *testing.T) {
+	testCases := []struct {
+		name     string
+		genState *types.GenesisState
+		expPanic bool
+	}{
+		{
+			name: "invalid issuers",
+			genState: &types.GenesisState{
+				Issuers: []*types.IssuerGenesisAccount{
+					{Address: "wrong address"},
+				},
+			},
+			expPanic: true,
+		},
+		{
+			name: "invalid issuers with no operators",
+			genState: &types.GenesisState{
+				Issuers: []*types.IssuerGenesisAccount{
+					{Address: "cosmos1qtu30xdvzkqxkluwpmacmluyxw23rw7ces8qtusn"},
+				},
+			},
+			expPanic: true,
+		},
+		{
+			name: "invalid issuer in verification data",
+			genState: &types.GenesisState{
+				VerificationDetails: []*types.GenesisVerificationDetails{
+					{
+						Id: hexutils.HexToBytes("0273FBBAFFC58F732199B20833643248C213C5DBA8F4A05DF505713FD36B8CE2"),
+						Details: &types.VerificationDetails{
+							IssuerAddress: "invalid issuer",
+						},
+					},
+				},
+			},
+			expPanic: true,
+		},
+		{
+			name: "invalid timestamp in verification data",
+			genState: &types.GenesisState{
+				VerificationDetails: []*types.GenesisVerificationDetails{
+					{
+						Id: hexutils.HexToBytes("0273FBBAFFC58F732199B20833643248C213C5DBA8F4A05DF505713FD36B8CE2"),
+						Details: &types.VerificationDetails{
+							IssuerAddress:       "cosmos1qyqs2z8qvvy6f3gm2f2hlpayje3wtcpfpnzqjn6pp6",
```

Comment:
Could you use swtr-prefixed addresses instead of cosmos

Reply:
Updated here: https://github.com/SigmaGmbH/swisstronik-chain/pull/73/commits/8c7ea9ea579f9f85e3b92bea018add026124bc57